pycircuitbreaker/pycircuitbreaker.py

```diff
@@ -160,13 +160,32 @@ def success_count(self) -> int:
         return self._strategy.success_count
 
 
+class CircuitBreakerRegistry:
+    def __init__(self) -> None:
+        self._registry: List[CircuitBreaker] = []
+
+    def register(self, circuitbreaker: CircuitBreaker) -> None:
```

Comment:
Would it make sense to have a bulk register endpoint as well? I'm thinking that would allow something similar to [Flask Blueprints](https://exploreflask.com/en/latest/blueprints.html) where there could be a registry for each module and then one overall registry that combines them?

I'm thinking it would enable something like

**views/module1.py**

```python
registry = CircuitBreakerRegister()

def db_breaker(func: Callable) -> Callable:
    breaker = CircuitBreaker(
        breaker_id="db",
        error_threshold=5,
        recovery_timeout=30,
        recovery_threshold=1,
        exception_blacklist=[DisconnectionError, TimeoutError],
    )
    registry.register(breaker)
    @wraps(func)
    def wrapper(*args: Any, **kwargs: Any) -> Any:
        return breaker.call(func, *args, **kwargs)
    return circuit_wrapper
@db_breaker
def call_to_db():
    ...
```

**app.py**

```python
from views.db import registry as db_registry

registry = CircuitBreakerRegistry()
registry.bulk_register(db_registry.get_circuits())
```

Reply:
I think it makes sense with endpoints because you potentially register a lot of them in a webapp so we need to find a way to organize code into layers.

We usually don't have that much external dependencies (a database, a message broker and few external services) so I didn't feel the need to propose a feature to support such layers but I can add it if you want to :)

Reply:
That makes sense; it can always be added later if needed 😄